File: modules/file-transfer-core/src/main/java/com/shterneregen/securelan/filetransfer/quickshare/QuickShareService.java
package com.shterneregen.securelan.filetransfer.quickshare;

import java.io.IOException;
import java.util.List;
import java.util.Optional;

public interface QuickShareService extends AutoCloseable {
    void start(QuickShareServerConfig config) throws IOException;

    void stop();

    boolean isRunning();

    int port();

    QuickShareSnapshot share(QuickShareCreateRequest request) throws IOException;

    Optional<QuickShareSnapshot> findShare(String id);

    List<QuickShareSnapshot> shares();

    boolean stopShare(String id);

    List<String> landingUrls();

    @Override
    default void close() {
        stop();
    }
}

File: modules/file-transfer-core/src/main/kotlin/com/shterneregen/securelan/filetransfer/quickshare/QuickShareService.kt
package com.shterneregen.securelan.filetransfer.quickshare

import java.io.IOException
import java.util.Optional

interface QuickShareService : AutoCloseable {
    @Throws(IOException::class)
    fun start(config: QuickShareServerConfig)

    fun stop()

    fun isRunning(): Boolean

    fun port(): Int

    @Throws(IOException::class)
    fun share(request: QuickShareCreateRequest): QuickShareSnapshot

    fun findShare(id: String): Optional<QuickShareSnapshot>

    fun shares(): List<QuickShareSnapshot>

    fun stopShare(id: String): Boolean

    fun landingUrls(): List<String>

    override fun close() = stop()
}
